bin/p2-inspect/main.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/armon/consul-api"
+	"github.com/square/p2/pkg/intent"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/reality"
+	"gopkg.in/alecthomas/kingpin.v1"
+)
+
+var (
+	consulUrl = kingpin.Flag("consul", "The hostname and port of a consul agent in the p2 cluster. Defaults to 0.0.0.0:8500.").String()
+	nodeName  = kingpin.Flag("node", "The node to inspect. By default, all nodes are shown.").String()
+	podId     = kingpin.Flag("pod", "The pod manifest ID to inspect. By default, all pods are shown.").String()
+)
+
+type NodePodStatus struct {
+	NodeName           string `json:"-"`
+	PodId              string `json:"-"`
+	IntentManifestSHA  string
+	RealityManifestSHA string
+	Health             NodeHealth
+}
+
+type NodeHealth struct {
+	Status string
+	Output string
+}
+
+func main() {
+	kingpin.Version("0.0.1")
+	kingpin.Parse()
+
+	client, err := consulapi.NewClient(&consulapi.Config{
+		Address: *consulUrl,
+	})
+	if err != nil {
+		log.Fatalf("Could not open consul client: %s", err)
+	}
+
+	intents, _, err := client.KV().List(intent.INTENT_TREE, nil)
+	if err != nil {
+		log.Fatalf("Could not list intent kvpairs: %s", err)
+	}
+	realities, _, err := client.KV().List(reality.REALITY_TREE, nil)
+	if err != nil {
+		log.Fatalf("Could not list reality kvpairs: %s", err)
+	}
+
+	statusMap := make(map[string]map[string]NodePodStatus)
+
+	for _, kvp := range intents {
```

Comment:
looks like a bit of repeated code when iterating over intents and reality. Worth combining?